### PR TITLE
story(VENLY-3967) - Retrieve HEDERA specific information about fees

### PR DESCRIPTION
**Pull request by ZhamakochyanErik**

### CODE REVIEW

File: provider-hedera/src/main/java/network/arkane/provider/hedera/mapper/CustomFeeMapper.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return HederaCustomFeeDto.builder()
+                                 .fractionalFees(fractionalFees)
+                                 .royaltyFees(royaltyFees)
+                                 .fixedFees(fixedFees)
+                                 .build();
+    }
+
+    private HederaFixedFee mapFixedFee(CustomFixedFee customFixedFee) {
+        return HederaFixedFee.builder()
+                             .feeCollectorAccountId(Optional.ofNullable(customFixedFee.getFeeCollectorAccountId()).map(Objects::toString).orElse(null))
```
Comment (maciejwitwicki) [inline provider-hedera/src/main/java/network/arkane/provider/hedera/mapper/CustomFeeMapper.java:44]: Logic for getting the account id is repeated three times so it should be extracted to method

File: provider-hedera/src/main/java/network/arkane/provider/hedera/nonfungible/HederaNonfungibleGateway.java [hunk 2]
```diff
@@ -97,35 +97,49 @@ public List<NftWalletDto> getNftWallets(final String walletAddress,
...
         if (nft == null) {
             return null;
         }
-        final NonFungibleAsset.NonFungibleAssetBuilder nonFungibleAssetBuilder = NonFungibleAsset.builder()
-                                                                                                 .id(nft.getSerialNumber().toString())
-                                                                                                 .contract(getNonFungibleContract(nft.getTokenId()));
+        HederaNonFungibleAsset.HederaNonFungibleAssetBuilder<?, ?> nonFungibleAssetBuilder = HederaNonFungibleAsset.builder()
+                                                                                                                   .id(nft.getSerialNumber().toString())
+                                                                                                                   .contract(getNonFungibleContract(nft.getTokenId()));
+        if (optionalTokenInfo.isPresent()) {
```
Comment (maciejwitwicki) [inline provider-hedera/src/main/java/network/arkane/provider/hedera/nonfungible/HederaNonfungibleGateway.java:126]: This new logic should be covered with an unit test

File: provider-hedera/src/main/java/network/arkane/provider/hedera/mapper/CustomFeeMapper.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import network.arkane.provider.token.HederaCustomFeeDto;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Component
+public class CustomFeeMapper {
```
Comment (maciejwitwicki) [inline provider-hedera/src/main/java/network/arkane/provider/hedera/mapper/CustomFeeMapper.java:20]: This class has some logic that could be covered with tests. Either I have missed them in this PR or they should be added ?

File: provider-api-model/src/main/java/network/arkane/provider/token/TokenInfo.java [hunk 4]
```diff
@@ -14,6 +14,8 @@ public class TokenInfo {
     private Integer decimals;
     private String type;
     private String logo;
+    private HederaCustomFeeDto customFees;
```
Comment (maciejwitwicki) [inline provider-api-model/src/main/java/network/arkane/provider/token/TokenInfo.java:17]: TokenInfo is used by all chains so this field being specific for Hedera seems to be odd. Maybe it should be named CustomFees what do you think?

**Review by maciejwitwicki — CHANGES_REQUESTED** [review]
General remark for this PR is that I would not make NonFungibleGateway interface generic just to have special fields for Hedera as generics introduce another level of complexity and are harder to understand and maintain. 

In this case that approach can be replaced with a single field added to the NonFungibleAsset - `boolean hasSenderTranferFee`  or something named similarly.

As I've read in the ticket description, our issue is that we don't want to allow selling of NFTs that have this "fallback fee" set. We then actually don't need to provide both `fallbackFees` and `feeScheduledKey` attributes to the wallet and market - as we won't even use them there. We can have just set `hasSenderTranferFee` set at the very early level in blockchain-providers and don't leak unnecessary attributes through other layers of our system. 

What do you think about it?

File: provider-hedera/src/main/java/network/arkane/provider/hedera/nonfungible/HederaNonfungibleGateway.java (hunk 2 above)
Comment (ZhamakochyanErik) [reply provider-hedera/src/main/java/network/arkane/provider/hedera/nonfungible/HederaNonfungibleGateway.java:128]: for this class is written integration test 
but I think I cannot test that logic during integration test

Comment (maciejwitwicki) [reply provider-hedera/src/main/java/network/arkane/provider/hedera/nonfungible/HederaNonfungibleGateway.java:128]: That integration test is Disabled.
You can test that logic during any test - the input with tokenInfo gives a different result from the `getNonFungible` method than without the tokenInfo.

**Review by maciejwitwicki — APPROVED** [review]

**Pull request merged**